src/main.test.ts

```diff
@@ -215,54 +281,93 @@ Elapsed time:  0 ms
+          await writeFile(
+            path.join(
+              repository.directoryPath,
+              '.yarn',
+              'plugins',
+              'some-plugin',
+            ),
+            'content for some-plugin',
+          );
+          await writeFile(
+            path.join(
+              repository.directoryPath,
+              '.yarn',
+              'releases',
+              'some-release',
+            ),
+            'content for some-release',
+          );
+          await writeFile(
+            path.join(repository.directoryPath, 'package.json'),
+            JSON.stringify({ packageManager: 'yarn' }),
+          );
+          await writeFile(
+            path.join(repository.directoryPath, 'README.md'),
+            'Install [Yarn whatever](...)',
           );
         }
-        const stdout = new MockWritable();
-        const stderr = new MockWritable();
+        const outputLogger = new FakeOutputLogger();
 
         await main({
           argv: ['node', 'module-lint'],
-          stdout,
-          stderr,
+          stdout: outputLogger.stdout,
+          stderr: outputLogger.stderr,
           config: {
             cachedRepositoriesDirectoryPath,
             defaultProjectNames: projectNames,
           },
         });
 
-        const output = stdout.data().map(stripAnsi).join('');
-
-        expect(output).toBe(
+        expect(outputLogger.getStderr()).toBe('');
+        expect(outputLogger.getStdout()).toBe(
           `
 repo-1
 ------
 
+- Is the classic Yarn config file (\`.yarnrc\`) absent? ✅
+- Does the package have a well-formed manifest (\`package.json\`)? ✅
+  - Does the \`packageManager\` field in \`package.json\` conform? ✅
+- Is \`README.md\` present? ✅
+  - Does the README conform by recommending the correct Yarn version to install? ✅
+- Are all of the files for Yarn Modern present, and do they conform? ✅
+  - Does the README conform by recommending the correct Yarn version to install? ✅
 - Does the \`src/\` directory exist? ✅
 
-Results:       1 passed, 0 failed, 1 total
+Results:       8 passed, 0 failed, 8 total
 Elapsed time:  0 ms
 
 
 repo-2
 ------
 
+- Is the classic Yarn config file (\`.yarnrc\`) absent? ✅
+- Does the package have a well-formed manifest (\`package.json\`)? ✅
+  - Does the \`packageManager\` field in \`package.json\` conform? ✅
+- Is \`README.md\` present? ✅
+  - Does the README conform by recommending the correct Yarn version to install? ✅
+- Are all of the files for Yarn Modern present, and do they conform? ✅
+  - Does the README conform by recommending the correct Yarn version to install? ✅
 - Does the \`src/\` directory exist? ✅
 
-Results:       1 passed, 0 failed, 1 total
+Results:       8 passed, 0 failed, 8 total
 Elapsed time:  0 ms
 
 `,
         );
       });
     });
 
-    it('lists the rules executed against the default repositories which fail', async () => {
+    it('lists the failing rules executed against the default repositories', async () => {
```

Comment:
Question, what is this test intended to do? Is it just demonstrating that it's possible for rules to fail? I'm wondering how it's different from the "passing" test; the only difference seems to be the number of failures.

Reply:
Hmm, good question. My thought here was that the "passing" test would exercise a maximal scenario, and the "failing" test would exercise a minimal scenario. That is, in the first case, all of the rules are satisfied, so all of them should be passing, and in the second case, none of them are satisfied, so none of them should be passing, and error messages should be shown for all of them. In both cases, we also test that if a rule has children and the rule fails, then none of the children are run.

Reply:
Oh I see. In that case, there is one test that is passing even in this "failing" scenario (the one about not having the old yarn v1 config around)

Reply:
Okay, good point. I guess instead of maximal/minimal what I really want is all-passing/all-failing. I've fixed the tests to honor this intent instead and renamed them to reference that better: [65c22a3](https://github.com/MetaMask/module-lint/pull/39/commits/65c22a3092a27d425ad43777bea829bec5e84fdd)